### ISSUE TITLE
Missing ember-fetch?

### ISSUE THREAD

**Issue by MichalBryxi** [empress/empress-blog#43]
1. New blog
1. Create new author
1. Create new post under that author
1. Run `ember s` 
1. Open browser on http://localhost:4200/
1. See:
```
Error: cannot find the `fetch` module or the `fetch` global. Did you mean to install the `ember-fetch` addon?
    at Module.callback (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/addon-tree-output/@ember-data/adapter/-private.js:151:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:106:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at requireModule (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:27:1)
    at resolveInitializer (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/addon-tree-output/ember-load-initializers/index.js:11:1)
    at registerInitializers (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/addon-tree-output/ember-load-initializers/index.js:28:1)
    at _default (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/addon-tree-output/ember-load-initializers/index.js:65:1)
    at Module.callback (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/unicorn-misa/app.js:13:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:106:1)
    at Module._reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:143:1)
    at Module.reify (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:130:1)
    at Module.exports (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:104:1)
    at Object.requireModule (/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-96526IRpvHI9WoTKO/out-388-broccoli_merge_trees/assets/vendor/loader/loader.js:27:1)
    at Object.<anonymous> (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/ember-app.js:210:18)
    at VMSandbox.run (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/vm-sandbox.js:18:15)
    at EmberApp.createEmberApp (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/ember-app.js:209:30)
    at EmberApp.retrieveSandboxedApp (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/ember-app.js:234:17)
    at new EmberApp (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/ember-app.js:61:21)
    at FastBoot._buildEmberApp (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/index.js:114:17)
    at new FastBoot (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/fastboot/src/index.js:52:10)
    at app.use (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli-fastboot/index.js:318:29)
    at Layer.handle [as handle_request] (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/express/lib/router/index.js:317:13)
    at /Users/mbryxi/pudr.com/unicorn-misa/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/express/lib/router/index.js:275:10)
    at err (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli/lib/tasks/server/middleware/broccoli-watcher/index.js:54:11)
    at watcher.then (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/broccoli-middleware/lib/watcher-middleware.js:35:7)
    at tryCatcher (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli/node_modules/rsvp/dist/lib/rsvp/-internal.js:39:19)
    at invokeCallback (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli/node_modules/rsvp/dist/lib/rsvp/-internal.js:211:31)
    at /Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli/node_modules/rsvp/dist/lib/rsvp/then.js:26:14
    at flush (/Users/mbryxi/pudr.com/unicorn-misa/node_modules/ember-cli/node_modules/rsvp/dist/lib/rsvp/asap.js:80:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
```

I created a repro repo for you: https://gitlab.com/michal.bryxi/unicorn-misa-pudr-com.git
You will want to switch to a branch: `bug/43-ember-fetch`

**Comment by MichalBryxi** [comment]
I wonder whether it might be because the project built with `ember-data 3.12` and as I [mentioned before](https://github.com/empress/empress-blog/pull/41) it looks like `v3.12` of `ember-data` complained about missing `ember-fetch`

**Comment by MichalBryxi** [comment]
Amendment: Fresh installation pulls in `"ember-data": "~3.11.0",`, but that has `"@ember-data/model": "^3.11.4",`, which inevitably at some level gets to `v3.12.x` of something.
I'm not really sure what is happening. Sorry.

**Comment by MichalBryxi** [comment]
Indeed `ember install ember-fetch` fixes the issue o_O

**Comment by MichalBryxi** [comment]
The error message mentioned here was introduced in: https://github.com/emberjs/data/pull/6155/files

**Comment by mansona** [comment]
@MichalBryxi sorry for the delay in getting back to you

This is something that is an upstream problem in ember-data, I just opened an issue here https://github.com/emberjs/data/issues/6326

for now, our workaround is to install ember-fetch as you said. If you look at the issue I linked above you will see that I refer to another issue with ember-data where they had a very similar issue. It is essentially a problem with the chain of dependencies.

Thanks for reporting this, I'll keep an eye on it for now and see if we can't rely on upstream to fix 👍 if not then we can always add ember-fetch to our blueprint so it will get installed when you install empress-blog
